spec-main/api-browser-window-spec.ts

```diff
@@ -1113,6 +1113,24 @@ describe('BrowserWindow module', () => {
           await unmaximize;
           expect(w.isMaximized()).to.equal(false);
         });
+        it('returns the correct value for windows with an aspect ratio', async () => {
+          w.destroy();
+          w = new BrowserWindow({
+            show: false,
+            fullscreenable: false
+          });
+
+          w.setAspectRatio(16 / 11);
+
+          const maximize = emittedOnce(w, 'resize');
+          w.show();
+          w.maximize();
+          await maximize;
+
+          expect(w.isMaximized()).to.equal(true);
+          w.resizable = false;
+          expect(w.isMaximized()).to.equal(true);
```

Comment:
The test here is incomplete because it does not cover the first question asked by https://github.com/electron/electron/issues/30979#issuecomment-919938987. It only covers the second question here.

Reply:
@BlackHole1 as noted below, this PR is not intended to fully close the issue. This test covers what is specifically being fixed here.

Reply:
ok